### ISSUE TITLE
support for repeatable annotations

### ISSUE THREAD

**Issue by yigit** [google/ksp#358]
Since Java 8, there is way to repeat annotations by providing a container class:

e.g.
``` java
@Repeatable(RepeatableJavaAnnotation.List.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface RepeatableJavaAnnotation {
    String value();
    @Retention(RetentionPolicy.RUNTIME)
    @interface List {
        RepeatableJavaAnnotation[] value();
    }
}
```

``` java
@RepeatableJavaAnnotation("x")
@RepeatableJavaAnnotation("y")
@RepeatableJavaAnnotation("z")
public class JavaSubject {}
```

When running with Java AP, you can get this list of values by querying with `RepeatableJavaAnnotation.List`.
That does not seem to be the case with KSP.

```
javap -v JavaSubject.class

SourceFile: "JavaSubject.java"
RuntimeVisibleAnnotations:
  0: #14(#15=[@#16(#15=s#17),@#16(#15=s#18),@#16(#15=s#19)])
    androidx.room.compiler.processing.testcode.RepeatableJavaAnnotation$List(
      value=[@androidx.room.compiler.processing.testcode.RepeatableJavaAnnotation(
        value="x"
      ),@androidx.room.compiler.processing.testcode.RepeatableJavaAnnotation(
        value="y"
      ),@androidx.room.compiler.processing.testcode.RepeatableJavaAnnotation(
        value="z"
      )]
    )

```

Normally, this could be considered OK assuming they are available in the annotations list (haven't verified but this is my guess). But when code is `compiled`, they do get moved into the `List` wrapper annotation above, which would make descriptor implementations inconsistent with declaration implementations (also possibly with PSI, haven't verified).

Btw, this might also have different behaviors between java target versions. I'm also hitting [this](https://youtrack.jetbrains.com/issue/KT-12794) issue even after i set target to 1.8. Will comment here once i find more information on why it is still happening.

**Comment by yigit** [comment]
I wanted to prototype this but it is not trivial at all.

java repeatable annotation is declared as:

```
public @interface Repeatable {
    Class<? extends Annotation> value();
}
```

whereas kotlin repeatable is:
```
@Target(AnnotationTarget.ANNOTATION_CLASS)
public annotation class MustBeDocumented
```

notice that the java api requires you to specify a container class whereas the kotlin one does not.

Now the issue is that, when java compiles a repeatable annotation; they actually wrap it in the container.
Kotlin [does not even support](https://youtrack.jetbrains.com/issue/KT-12794) non-source retention repeatable annotations for java (didn't try w/ kotlin annotation yet, i assume it won't).

In KSP right now, if you are reading from a `origin=class`, you need to use the container whereas if you are reading from origin `java, kotlin`, you need to use the repeated annotation.

I wanted to send a PR where KSP does the wrapping as well to be consistent but I cant because i cannot read the target out of `Repeatable` (it does not have the container value :/ )

so tl;dr; i'm not even sure how we can fix this in KSP :'(
We could consider unwrapping the annotations in `.class` files but we cannot do that because there is no reference from the container annotation to the repeated annotation.
